### PR TITLE
pl/custom dashboards

### PR DESCRIPTION
**Pull request by tlindsay**
Adds support for [Fastly Custom Observability Dashboards](https://www.fastly.com/documentation/reference/api/observability/custom-dashboards/) CRUD operations

### CODE REVIEW

File: examples/resources/custom_dashboard_basic_usage.tf [hunk 1]
```diff
@@ -0,0 +1,251 @@
+resource "fastly_custom_dashboard" "example" {
+  name        = "Example Custom Dashboard"
+  description = "This is an example custom dashboard. A few dashboard items are been provided to help you get started."
```
Comment (darin-nee) [inline examples/resources/custom_dashboard_basic_usage.tf]: Minor typo with `are been`

File: examples/resources/custom_dashboard_basic_usage.tf [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        format             = "percent"
+        plot_type          = "donut"
+        calculation_method = "latest"
+      }
+    }
+  }
+
+  dashboard_item {
+    title    = "Client & Server Errors"
+    subtitle = "Total errors served from the client of server."
```
Comment (darin-nee) [inline examples/resources/custom_dashboard_basic_usage.tf]: s/of/or/

File: fastly/resource_fastly_custom_dashboard.go [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+				MinItems:    0,
+				MaxItems:    100,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"data_source": &schemaDataSource,
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Dashboard item identifier (UUID).",
+						},
```
Comment (darin-nee) [inline fastly/resource_fastly_custom_dashboard.go]: Does this need to be exposed?

**Review by darin-nee — COMMENTED** [review]
A non-blocking question and a couple of minor typos otherwise this looks great to me.

**Review by darin-nee — APPROVED** [review]

**Comment by kpfleming** [comment]
This needs to be reviewed by a member of the DevTools team, who build and support the TF provider, before it gets merged.

**Comment by tlindsay** [comment]
> This needs to be reviewed by a member of the DevTools team, who build and support the TF provider, before it gets merged.

Sorry, @kpfleming. I looked through the CONTRIBUTING.md and DEVELOPMENT.md and didn't see any mention of that part of the process and Github didn't seem to have any rules, so I assumed that it was alright to merge.

How would you like to handle this? Should I revert?

**Pull request merged**